asr1k_neutron_l3/plugins/db/asr1k_db.py

```diff
@@ -649,6 +649,17 @@ def set_router_dynamic_nat_pool(self, context, router_id, dynamic_nat_pool):
             ra = self.get_router_att(context, router_id)
             ra.dynamic_nat_pool = dynamic_nat_pool
 
+    @db_api.CONTEXT_READER
+    def get_subnet_address_scope_map(self, context, subnet_ids):
+        query = context.session.query(models_v2.Subnet.id,
+                                      models_v2.SubnetPool.address_scope_id)
+        query = query.outerjoin(
```

Comment:
why "outer" but not "inner"? (what case do we want to get from an outer join that we don't get with the inner?

Reply:
We do an outer join so we get a result for each subnet, even if it has no subnet pool attached, as in these cases we want to have `None` as address scope, i.e. subnet exists in dict, address scope `None`.

Reply:
Ok, so `None` has also a meaning that we use, that is what i needed to know (also answered in another comment). Thanks!